File: Segundo/PDOO/java/juegoTexto/Controlador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package juegoTexto;

import civitas.*;


public class Controlador {
    private CivitasJuego juego;
    private VistaTextual vista;
    
    Controlador(CivitasJuego refJuego, VistaTextual refVista) {
        juego = refJuego;
        vista = refVista;
    }
    
    void juega() {
        vista.setCivitasJuego(juego);
        Boolean fin = juego.finalDelJuego();
        OperacionesJuego operacion;
        while (!fin) {
            vista.actualizarVista();
            vista.pausa();
            operacion = juego.siguientePaso();
            vista.mostrarSiguienteOperacion(operacion);
            if (operacion != OperacionesJuego.PASAR_TURNO) {
                vista.mostrarEventos();
            }
            fin = juego.finalDelJuego();
            if (!fin) switch (operacion) {
                case COMPRAR:
                    Respuestas respuesta = vista.comprar();
                    if (respuesta == Respuestas.SI) {
                        if (juego.comprar())
                            juego.siguientePasoCompletado(operacion);                            
                    } else
                        juego.siguientePasoCompletado(operacion);
                    break;
                case GESTIONAR:
                    vista.gestionar();
                    int iGestion = vista.getGestion();
                    int iPropiedad = vista.getPropiedad();
                    GestionesInmobiliarias gestion = GestionesInmobiliarias.values()[iGestion];
                    OperacionInmobiliaria nuevaOperacion = new OperacionInmobiliaria(gestion, iPropiedad);
                    if (null != gestion) switch (gestion) {
                        case VENDER:
                            juego.vender(iPropiedad);
                            break;
                        case HIPOTECAR:
                            juego.hipotecar(iPropiedad);
                            break;
                        case CANCELAR_HIPOTECA:
                            juego.cancelarHipoteca(iPropiedad);
                            break;
                        case CONSTRUIR_CASA:
                            juego.construirCasa(iPropiedad);
                            break;
                        case CONSTRUIR_HOTEL:
                            juego.construirHotel(iPropiedad);
                            break;
                        case TERMINAR:
                            juego.siguientePasoCompletado(operacion);
                            break;
                        default:
                            break;
                    }
                    break;
                case SALIR_CARCEL:
                    SalidasCarcel salida = vista.salirCarcel();
                    if (salida == SalidasCarcel.PAGANDO) {
                        juego.salirCarcelPagando();
                    } else {
                        juego.salirCarcelTirando();
                    }
                    juego.siguientePasoCompletado(operacion);
                    break;
                default:
                    break;                    
            }
        }
        vista.ranking();
    }
    
    void setDebug(Boolean d) {
        Dado.getInstance().setDebug(d);
    }
    
}

File: Segundo/PDOO/CivitasRuby/Civitas/Controlador.rb
# encoding:utf-8

require_relative 'CivitasJuego'
require_relative 'vista_textual'
require_relative 'Respuestas'
require_relative 'GestionesInmobiliarias'
require_relative 'OperacionInmobiliaria'

module Civitas

    class Controlador
        


        def initialize(civitas, vis)
            @juego = civitas
            @vista = vis
        end

        def juega

            @vista.setCivitasJuego(@juego)
            fin = @juego.finalDelJuego()
            while ( !fin )

                @vista.actualizarVista()
                @vista.pausa()

                siguientepaso = @juego.siguientePaso()

                @vista.mostrarSiguienteOperacion(siguientepaso)

                if (siguientepaso != Operaciones_juego::PASAR_TURNO)
                    @vista.mostrarEventos()
                end

                fin = @juego.finalDelJuego()

                if !fin
                    
                    case siguientepaso
                    when Operaciones_juego::COMPRAR

                        respuesta = @vista.comprar()
                        if respuesta == Respuestas::SI
                            @juego.comprar
                        end
                        @juego.siguientePasoCompletado(siguientepaso)

                    when Operaciones_juego::GESTIONAR
                        lista_Gestiones = [GestionesInmobiliarias::VENDER, GestionesInmobiliarias::HIPOTECAR, GestionesInmobiliarias::CANCELAR_HIPOTECA, GestionesInmobiliarias::CONSTRUIR_CASA, GestionesInmobiliarias::CONSTRUIR_HOTEL, GestionesInmobiliarias::TERMINAR]

                        @vista.gestionar()
                        indiceg = @vista.getGestion()
                        indicep = @vista.getPropiedad()
                        operacion = OperacionInmobiliaria.new(indiceg, indicep)

                        case lista_Gestiones[indiceg]

                        when GestionesInmobiliarias::VENDER
                            @juego.vender(indicep)
                            @juego.siguientePasoCompletado(siguientepaso)

                        when GestionesInmobiliarias::HIPOTECAR
                            @juego.hipotecar(indicep)
                            @juego.siguientePasoCompletado(siguientepaso)

                        when GestionesInmobiliarias::CANCELAR_HIPOTECA
                            @juego.cancelarHipoteca(indicep)
                            @juego.siguientePasoCompletado(siguientepaso)

                        when GestionesInmobiliarias::CONSTRUIR_CASA
                            @juego.construirCasa(indicep)
                            @juego.siguientePasoCompletado(siguientepaso)

                        when GestionesInmobiliarias::CONSTRUIR_HOTEL
                            @juego.construirHotel(indicep)
                            @juego.siguientePasoCompletado(siguientepaso)

                        when GestionesInmobiliarias::TERMINAR
                            @juego.siguientePasoCompletado(siguientepaso)
                        end
                        
                    when Operaciones_juego::SALIR_CARCEL

                        salidaCarcel = @vista.salirCarcel()
                        if salidaCarcel == SalidasCarcel::PAGANDO
                            @juego.salirCarcelPagando()
                        else
                            @juego.salirCarcelTirando()
                        end

                    end
                end
            end

            @vista.ranking(9)

        end

    end

end